aten/src/ATen/Dispatch.h

```diff
@@ -36,6 +36,17 @@ struct ScalarTypeToCType<at::ScalarType::Half> {
   static at::Half t;
 };
 
+template<>
+struct ScalarTypeToCType<at::ScalarType::BFloat16> {
+  using type = at::BFloat16;
+
+  // This is a workaround for the CUDA bug which prevents ::detail::ScalarTypeToCType<T>::type being used directly
+  // due to ambiguous reference which can't to be resolved. For some reason it cant pick between at::detail and at::cuda::detail.
+  // For repro example, please see: https://gist.github.com/izdeby/952ae7cf256ddb740a73776d39a7e7ba
```

Comment:
Did we report this upstream to NVIDIA? Once you have an account, you can file a bug at https://developer.nvidia.com/nvidia_bug/add

Reply:
I talked to @ngimel and she told me that she will file a bug which should have been fixed in cuda 10.2, but as we still have to support older versions of cuda, this hack is here to stay